spec/support/shared/api_client.rb

```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
-RSpec.shared_context 'API client', shared_context: :metadata do
+RSpec.shared_context 'API client', shared_context: :metadata do |params|
   before do
     OpenWeather::Config.reset
+    OpenWeather::Config.endpoint = params[:endpoint] if params&.key?(:endpoint)
```

Comment:
For a future PR, I'd like to make it future proof able to write `OpenWeather::Client.configure(params)` here.